lib/std/math.zig

```diff
@@ -184,6 +184,24 @@ test "approxEqAbs and approxEqRel" {
         try testing.expect(approxEqAbs(T, min_value, 0.0, eps_value * 2));
         try testing.expect(approxEqAbs(T, -min_value, 0.0, eps_value * 2));
     }
+
+    comptime {
+        // `comptime_float` is guaranteed to have the same precision and operations of
+        // the largest other floating point type, which is f128 but it doesn't have a
+        // defined layout so we can't rely on `@bitCast` to construct the smallest
+        // possible epsilon value like we do in the tests above. In the same vein, we
+        // also can't represent a max/min, `NaN` or `Inf` values.
+        const eps_value = 1e-4;
```

Comment:
1. See https://en.wikipedia.org/wiki/Machine_epsilon for the values of epsilon of only the number representation. https://docs.oracle.com/cd/E19957-01/806-3568/ncg_goldberg.html and IEEE754 math routines for mandatory precision.
2. `comptime_float` has infinite precision due to usage of arbitrary precision floats, so I do not understand where you get f128 from or how you derived that it is f128. The language reference is rather conservative regarding guarantees and does try not to introduce implementation details (here I think for the worse). Afaiu, applying both functions makes no sense for `comptime_float` due to arbitrary precision arithmetic usage. Yes, comptime float literals are guaranteed to be at least f128 precision https://ziglang.org/documentation/master/#Float-Literals, but this is (afaiu) due to implicit coercion.
3. f128 epsilon for operations should be much much smaller than `10^(-4)`
4. This sounds like the behavior for NaN and +-inf diverges between comptime_float and the other floats, because `>0/0.0` is inf, `<0/0.0` is -inf and `0.0/0.0` is NaN but a compile error at comptime.

Reply:
The `comptime_float`/`f128` comment is from the [Zig docs](https://ziglang.org/documentation/0.11.0/#toc-Float-Literals) as you also found.

I chose `10e-4` arbitrarily since choosing any small number isn't suitable for `comptime_float` given it's infinite precision. I can update to whatever we think is suitable though.

> Afaiu, applying both functions makes no sense for comptime_float due to arbitrary precision arithmetic usage.

My use case is to be able to do a sanity check like the following. Good to sanity check because the foot-gun here is `comptime_int` division and the result being cast/rounded to `0` unexpectedly.

```zig
const h_raw_lower_bound = -1;
const h_raw_upper_bound = 5;
const h_raw_range: comptime_float = h_raw_upper_bound - h_raw_lower_bound;

const h_normalized_upper_bound: comptime_float = 1;
const h_normalized_scaler: comptime_float = h_normalized_upper_bound / h_raw_range;

// Sanity check
std.debug.assert(approxEqAbs(comptime_float, h_normalized_scaler, 0.1666666, 1e-4));
```

Reply:
docs say `Float literals have type comptime_float which is guaranteed to have the same precision and operations of the largest other floating point type, which is f128.` where are you getting that they have infinite precision? it says `the same precision`, not at least